docs/img-src/architecture.mmd

```diff
@@ -0,0 +1,11 @@
+%% Ca( for flowchart below
```

Comment:
For me it helped my understand which parts of the system even exist in detox and what their responsibilities are, but I can remove it if you don't like it